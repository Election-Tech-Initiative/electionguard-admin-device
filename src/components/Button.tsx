import styled, { css } from 'styled-components'

interface Attrs extends HTMLButtonElement {
  readonly type: string
}

export interface ButtonInterface {
  readonly danger?: boolean
  readonly primary?: boolean
  readonly fullWidth?: boolean
  readonly small?: boolean
  readonly big?: boolean
}

interface Props
  extends ButtonInterface,
    React.PropsWithoutRef<JSX.IntrinsicElements['button']> {}

const buttonStyles = css<Props>`
  border: none;
  border-radius: 0.25rem;
  box-sizing: border-box;
  background: ${({ danger = false, primary = false }) =>
    (danger && 'red') ||
    (primary && 'rgb(71, 167, 75)') ||
    'rgb(211, 211, 211)'};
  cursor: ${({ disabled = false }) => (disabled ? undefined : 'pointer')};
  width: ${({ fullWidth = false }) => (fullWidth ? '100%' : undefined)};
  padding: ${({ big = false, small = false }) =>
    // eslint-disable-next-line no-nested-ternary
    small ? '0.35rem 0.5rem' : big ? '1rem 1.75rem' : '0.75rem 1rem'};
  line-height: 1.25;
  color: ${({ disabled = false, danger = false, primary = false }) =>
    (disabled && 'rgb(169, 169, 169)') ||
    (danger && '#FFFFFF') ||
    (primary && '#FFFFFF') ||
    'black'};
  font-size: ${({ big = false }) => (big ? '1.25rem' : undefined)};
`

export const DecoyButton = styled.div`
  ${buttonStyles} /* stylelint-disable-line value-keyword-case */
`

const Button = styled('button').attrs((props: Attrs) => ({
  type: props.type || 'button',
}))`
  ${buttonStyles} /* stylelint-disable-line value-keyword-case */
`

export default Button
