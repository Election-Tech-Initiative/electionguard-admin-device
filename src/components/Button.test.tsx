import React from 'react'
import { render } from '@testing-library/react'

import Button, { DecoyButton } from './Button'

it(`renders Button`, () => {
  const { container } = render(<Button>foo</Button>)
  expect(container.firstChild).toMatchSnapshot()
})

it(`renders primary Button`, () => {
  const { container } = render(<Button primary>Primary</Button>)
  expect(container.firstChild).toMatchSnapshot()
})

it(`renders danger Button`, () => {
  const { container } = render(<Button danger>Danger!</Button>)
  expect(container.firstChild).toMatchSnapshot()
})

it(`renders big Button`, () => {
  const { container } = render(<Button big>I’m a big button!</Button>)
  expect(container.firstChild).toMatchSnapshot()
})

it(`renders small Button`, () => {
  const { container } = render(<Button small>I’m a small button!</Button>)
  expect(container.firstChild).toMatchSnapshot()
})

it(`renders DecoyButton`, () => {
  const { container } = render(<DecoyButton>DecoyButton</DecoyButton>)
  expect(container.firstChild).toMatchSnapshot()
})
