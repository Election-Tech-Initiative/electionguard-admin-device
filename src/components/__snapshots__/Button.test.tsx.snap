// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Button 1`] = `
<button
  class="sc-bwzfXH kHZqpR"
  type="button"
>
  foo
</button>
`;

exports[`renders DecoyButton 1`] = `
<div
  class="sc-bdVaJa dnJkEm"
>
  DecoyButton
</div>
`;

exports[`renders big Button 1`] = `
<button
  class="sc-bwzfXH eYHDXc"
  type="button"
>
  I’m a big button!
</button>
`;

exports[`renders danger Button 1`] = `
<button
  class="sc-bwzfXH kDljbx"
  type="button"
>
  Danger!
</button>
`;

exports[`renders primary Button 1`] = `
<button
  class="sc-bwzfXH URjIp"
  type="button"
>
  Primary
</button>
`;

exports[`renders small Button 1`] = `
<button
  class="sc-bwzfXH jhNkEE"
  type="button"
>
  I’m a small button!
</button>
`;
